include/opw_kinematics/opw_kinematics_impl.h

```diff
@@ -116,17 +213,103 @@ Solutions<T> inverse(const Parameters<T>& params, const Transform<T>& pose) noex
   T theta5_vii = -theta5_iii;
   T theta5_viii = -theta5_iv;
 
-  T theta6_i = std::atan2(matrix(0, 1) * s23[0] * cos1[0] + matrix(1, 1) * s23[0] * sin1[0] + matrix(2, 1) * c23[0],
-                          -matrix(0, 0) * s23[0] * cos1[0] - matrix(1, 0) * s23[0] * sin1[0] - matrix(2, 0) * c23[0]);
-
-  T theta6_ii = std::atan2(matrix(0, 1) * s23[1] * cos1[1] + matrix(1, 1) * s23[1] * sin1[1] + matrix(2, 1) * c23[1],
-                           -matrix(0, 0) * s23[1] * cos1[1] - matrix(1, 0) * s23[1] * sin1[1] - matrix(2, 0) * c23[1]);
-
-  T theta6_iii = std::atan2(matrix(0, 1) * s23[2] * cos1[2] + matrix(1, 1) * s23[2] * sin1[2] + matrix(2, 1) * c23[2],
-                            -matrix(0, 0) * s23[2] * cos1[2] - matrix(1, 0) * s23[2] * sin1[2] - matrix(2, 0) * c23[2]);
+  T zero_threshold{ T(1e-6) };
+  T theta4_i, theta6_i;
+  if (std::abs(theta5_i) < zero_threshold)
+  {
+    theta4_i = 0;
+    Vector xe(matrix(0, 0), matrix(1, 0), matrix(2, 0));
+    Matrix3 Rc;
+    Rc.col(1) = Vector(-std::sin(theta1_i), std::cos(theta1_i), 0);  // yc
+    Rc.col(2) = Vector(matrix(0, 2), matrix(1, 2), matrix(2, 2));    // zc and ze are equal
+    Rc.col(0) = Rc.col(1).cross(Rc.col(2));                          // xc
```

Comment:
This code has never won any awards for readability, but still I'll nitpick on the mixed usage of `col(int)` and explicitly writing out `matrix(0, 2), matrix(1, 2) ...` in the same block of code. Perhaps converge it all to one (maybe the col one)?

Reply:
Are suggesting this instead?
``` c++
    Matrix3 Rc;

    // yc
    Rc(0,1) = -std::sin(theta1_i);
    Rc(1,1) = std::cos(theta1_i);
    Rc(2,1) = 0;

    // zc and ze are equal
    Rc(0, 2) = matrix(0, 2);
    Rc(1, 2) = matrix(1, 2);
    Rc(2, 2) = matrix(2, 2);

    // xc
    Vector xc = Rc.col(1).cross(Rc.col(2));
    Rc(0, 0) = xc(0);
    Rc(1, 0) = xc(1);
    Rc(2, 0) = xc(2);
```

Reply:
Sorry for the unclear comment; I was suggesting:
```C++
   //  Rc.col(2) = Vector(matrix(0, 2), matrix(1, 2), matrix(2, 2));    // zc and ze are equal 
   // goes to
   // Rc.col(2) = matrix.col(2);    
```